Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
+    public convenience init<S: AsyncSequence>(
+        sequence: S,
+        length: Body.Length,
+        iterationBehavior: IterationBehavior
+    ) where S.Element: StringProtocol {
+        self.init(
+            sequence: .init(sequence.map(\.asBodyChunk)),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+}
+
+extension Body {
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes`, converts it to String, and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsString(upTo maxBytes: Int) async throws -> String {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return String(decoding: bytes, as: UTF8.self)
+    }
+}
+
+// MARK: - Body conversions
+
+extension Body: ExpressibleByStringLiteral {
+
+    public convenience init(stringLiteral value: String) {
+        self.init(data: value)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: [UInt8]) {
+        self.init(data: data[...])
+    }
+}
+
+extension Body: ExpressibleByArrayLiteral {
+
+    public typealias ArrayLiteralElement = UInt8
+
+    public convenience init(arrayLiteral elements: UInt8...) {
+        self.init(data: elements)
+    }
+}
+
+extension Body {
+
+    public convenience init(data: Data) {
+        self.init(data: ArraySlice(data))
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes`, converts it to Foundation.Data, and returns it.
+    /// - Parameters:
+    ///   - maxBytes: The maximum number of bytes this method is allowed
+    ///     to accumulate in memory before it throws an error.
+    /// - Throws: `TooManyBytesError` if the the body contains more
+    ///   than `maxBytes`.
+    public func collectAsData(upTo maxBytes: Int) async throws -> Data {
+        let bytes: DataType = try await collect(upTo: maxBytes)
+        return Data(bytes)
+    }
+}
+
+// MARK: - Underlying async sequences
+
+extension Body {
+
+    /// Async iterator of both input async sequences and of the body itself.
+    public struct Iterator: AsyncIteratorProtocol {
+
+        public typealias Element = Body.DataType
+
+        private let produceNext: () async throws -> Element?
+
+        init<Iterator: AsyncIteratorProtocol>(
+            _ iterator: Iterator
+        ) where Iterator.Element == Element {
+            var iterator = iterator
+            self.produceNext = {
+                try await iterator.next()
+            }
```

Comment:
Why are we not storing the iterator here but a closure?

Reply:
The intent here is type-erasure, is there a better way than to go through a closure?

Reply:
Even if `AsyncIteratorProtocol` had primary associated type, we would prefer a closure as it will get specialised but an existential not, as of today.